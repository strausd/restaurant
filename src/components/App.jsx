import React from 'react';
import { connect } from 'react-redux';

import AppRouter from '../routers/AppRouter';
import { fetchRestaurants } from '../actions/baseActions'


class App extends React.Component {
    constructor(props) {
        super(props);
        props.fetchRestaurants();
    }

    render() {
        return (
            <div>
                <AppRouter />
            </div>
        );
    }
}

const mapDispatchToProps = dispatch => {
    return {
        fetchRestaurants: () => dispatch(fetchRestaurants())
    };
};

export default connect(undefined, mapDispatchToProps)(App);