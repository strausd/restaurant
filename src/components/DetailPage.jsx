import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import RestaurantDetailItem from './RestaurantDetailItem';


export class DetailPage extends React.Component {
    render() {
        const id = this.props.match.params.id;
        if (isNaN(id)) {
            return <Redirect to="/" />
        } else if (!this.props.restaurants || this.props.restaurants.length < 1) {
            return (
                <div className="page-content">
                    Loading...
                </div>
            );
        } else {
            const restaurant = this.props.restaurants[id];
            return (
                <div className="page-content">
                    <RestaurantDetailItem restaurant={restaurant} />
                </div>
            );
        }
    }
}


const mapStateToProps = ({ restaurants, restaurant }) => {
    return {
        restaurants,
        restaurant
    };
};

export default connect(mapStateToProps)(DetailPage);