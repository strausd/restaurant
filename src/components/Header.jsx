import React from 'react';
import { Link } from 'react-router-dom';


const Header = props => {
    const back_button = window.location.pathname !== '/';
    return (
        <div className="header">
            {back_button ? (
                <div className="header__icon header__back">
                    <Link to="/">
                        <img src="/dist/icons/ic_webBack@2x.png" alt="Map" />
                    </Link>
                </div>
            ) : null}
            <div className="header__link">
                <Link to="/">
                    <h1>Lunch Tyme</h1>
                </Link>
            </div>
            <div className="header__icon header__map">
                <Link to="/map">
                    <img src="/dist/icons/icon_map@2x.png" alt="Map" />
                </Link>
            </div>
        </div>
    );
};

export default Header;