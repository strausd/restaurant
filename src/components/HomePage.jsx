import React from 'react';
import { connect } from 'react-redux'

import RestaurantsList from './RestaurantsList';


export const HomePage = () => {
    return (
        <div className="page-content">
            <RestaurantsList />
        </div>
    );
};

export default HomePage;