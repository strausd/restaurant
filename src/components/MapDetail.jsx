import React from 'react';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps"

const MapDetail = withScriptjs(withGoogleMap((props) =>
    <GoogleMap
        defaultZoom={props.zoom}
        defaultCenter={{ lat: props.lat, lng: props.lng }}
    >
        <Marker position={{ lat: props.lat, lng: props.lng }}>
            <InfoWindow>
                <div>Testing</div>
            </InfoWindow>
        </Marker>
    </GoogleMap>
));

export default MapDetail;