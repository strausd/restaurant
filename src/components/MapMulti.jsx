import React from 'react';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps";

export const MapDetail = withScriptjs(withGoogleMap((props) =>
    <GoogleMap
        defaultZoom={props.zoom}
        defaultCenter={{ lat: props.lat, lng: props.lng }}
    >
        {props.restaurants.map((r, index) => {
            return (
                <Marker key={index} position={{ lat: r.lat, lng: r.lng }} onClick={() => props.onMarkerToggle(index)}>
                    {r.isOpenMarkerInfo && <InfoWindow onCloseClick={props.onMarkerToggle}>
                        <div className="marker-info">
                            <div className="marker-info__name">{r.name}</div>
                            <div className="marker-info__directions">
                                <a target="_blank" href={'https://www.google.com/maps/dir/?api=1&destination=' + r.gMapsQuery}>
                                    Get Directions
                                </a>
                            </div>
                        </div>
                    </InfoWindow>}
                </Marker>
            );
        })}
    </GoogleMap>
));

export default MapDetail;