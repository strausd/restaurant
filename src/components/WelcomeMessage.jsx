import React from 'react';
import { TransitionGroup, CSSTransition, Transition } from "react-transition-group";


export class WelcomeMessage extends React.Component {
    message = () => (
        <div className="starter hidden-mobile">
            <h1>Welcome to Lunch Tyme</h1>
            <h3>🍕🍔🌮</h3>
            <h3>Step 1 - Browse</h3>
            <h3>Step 2 - Go</h3>
            <h3>Step 3 - Eat</h3>
        </div>
    );

    render() {
        return (
            <CSSTransition in={this.props.show} classNames="fade" timeout={250} mountOnEnter unmountOnExit>
                {this.message()}
            </CSSTransition>
        );
    }
}

export default WelcomeMessage;