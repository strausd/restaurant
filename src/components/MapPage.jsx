import React from 'react';
import { connect } from 'react-redux';

import MapMulti from './MapMulti';
import { getMapInfoArray } from '../misc/utils';


export class MapPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            restaurant_marker_info: []
        };
    }

    componentWillMount() {
        if (!this.props.restaurants || this.props.restaurants.length < 1) {
            this.setState({ restaurant_marker_info: [] });
        } else {
            const restaurants = getMapInfoArray(this.props.restaurants);
            this.setState({ restaurant_marker_info: restaurants });
        }
    }
    

    componentWillReceiveProps(nextProps) {
        if (!nextProps.restaurants || nextProps.restaurants.length < 1) {
            this.setState({ restaurant_marker_info: [] });
        } else {
            const restaurants = getMapInfoArray(nextProps.restaurants);
            this.setState({ restaurant_marker_info: restaurants });
        }
    }

    onMarkerToggle = (index) => {
        this.setState(prevState => {
            return {
                restaurant_marker_info: prevState.restaurant_marker_info.map((r, r_index) => {
                    if (index === r_index) {
                        return {
                            ...r,
                            isOpenMarkerInfo: !r.isOpenMarkerInfo
                        };
                    } else {
                        return {
                            ...r,
                            isOpenMarkerInfo: false
                        };
                    }
                })
            };
        });
    }

    render() {
        if (!this.props.restaurants || this.props.restaurants.length < 1) {
            return (
                <div className="page-content">
                    Loading...
                </div>
            );
        } else {
            return (
                <div className="page-content">
                    <div className="map-page">
                        <MapMulti 
                            isMarkerShow={true}
                            googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyBGyVswgcCkctFs8x4X-14BS16VXWINHPA&v=3.exp&libraries=geometry,drawing,places"
                            loadingElement={<div style={{ height: `100%` }} />}
                            containerElement={<div style={{ height: `100%` }} />}
                            mapElement={<div style={{ height: `100%` }} />}
                            lat={32.98}
                            lng={-96.835}
                            zoom={13}
                            onMarkerToggle={this.onMarkerToggle}
                            restaurants={this.state.restaurant_marker_info}
                        />
                    </div>
                </div>
            );
        }
        
    }
}

const mapStateToProps = ({ restaurants }) => {
    return {
        restaurants
    };
};

export default connect(mapStateToProps)(MapPage);