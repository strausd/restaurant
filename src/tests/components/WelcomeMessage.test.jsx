import React from 'react';
import { shallow } from 'enzyme';

import { WelcomeMessage } from '../../components/WelcomeMessage';


test('should render WelcomeMessage component properly', () => {
    const wrapper = shallow(<WelcomeMessage />);
    expect(wrapper).toMatchSnapshot();
});