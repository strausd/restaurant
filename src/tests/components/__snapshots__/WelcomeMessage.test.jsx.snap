// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WelcomeMessage component properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WelcomeMessage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="starter hidden-mobile"
      >
        <h1>
          Welcome to Lunch Tyme
        </h1>
        <h3>
          🍕🍔🌮
        </h3>
        <h3>
          Step 1 - Browse
        </h3>
        <h3>
          Step 2 - Go
        </h3>
        <h3>
          Step 3 - Eat
        </h3>
      </div>,
      "classNames": "fade",
      "in": undefined,
      "mountOnEnter": true,
      "timeout": 250,
      "unmountOnExit": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome to Lunch Tyme
          </h1>,
          <h3>
            🍕🍔🌮
          </h3>,
          <h3>
            Step 1 - Browse
          </h3>,
          <h3>
            Step 2 - Go
          </h3>,
          <h3>
            Step 3 - Eat
          </h3>,
        ],
        "className": "starter hidden-mobile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Lunch Tyme",
          },
          "ref": null,
          "rendered": "Welcome to Lunch Tyme",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "🍕🍔🌮",
          },
          "ref": null,
          "rendered": "🍕🍔🌮",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Step 1 - Browse",
          },
          "ref": null,
          "rendered": "Step 1 - Browse",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Step 2 - Go",
          },
          "ref": null,
          "rendered": "Step 2 - Go",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Step 3 - Eat",
          },
          "ref": null,
          "rendered": "Step 3 - Eat",
          "type": "h3",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="starter hidden-mobile"
        >
          <h1>
            Welcome to Lunch Tyme
          </h1>
          <h3>
            🍕🍔🌮
          </h3>
          <h3>
            Step 1 - Browse
          </h3>
          <h3>
            Step 2 - Go
          </h3>
          <h3>
            Step 3 - Eat
          </h3>
        </div>,
        "classNames": "fade",
        "in": undefined,
        "mountOnEnter": true,
        "timeout": 250,
        "unmountOnExit": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Welcome to Lunch Tyme
            </h1>,
            <h3>
              🍕🍔🌮
            </h3>,
            <h3>
              Step 1 - Browse
            </h3>,
            <h3>
              Step 2 - Go
            </h3>,
            <h3>
              Step 3 - Eat
            </h3>,
          ],
          "className": "starter hidden-mobile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Lunch Tyme",
            },
            "ref": null,
            "rendered": "Welcome to Lunch Tyme",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "🍕🍔🌮",
            },
            "ref": null,
            "rendered": "🍕🍔🌮",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Step 1 - Browse",
            },
            "ref": null,
            "rendered": "Step 1 - Browse",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Step 2 - Go",
            },
            "ref": null,
            "rendered": "Step 2 - Go",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Step 3 - Eat",
            },
            "ref": null,
            "rendered": "Step 3 - Eat",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
